tests/override_config_table/test_override_config_table.py

```diff
@@ -0,0 +1,155 @@
+import json
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+
+GOLDEN_CONFIG = "/etc/sonic/golden_config_db.json"
+GOLDEN_CONFIG_BACKUP = "/etc/sonic/golden_config_db.json_before_override"
+CONFIG_DB = "/etc/sonic/config_db.json"
+CONFIG_DB_BACKUP = "/etc/sonic/config_db.json_before_override"
+
+logger = logging.getLogger(__name__)
+
+
+def file_exists_on_dut(duthost, filename):
+    return duthost.stat(path=filename).get('stat', {}).get('exists', False)
+
+
+@pytest.fixture(scope="module")
+def golden_config_exists_on_dut(duthost):
+    return file_exists_on_dut(duthost, GOLDEN_CONFIG)
+
+
+def backup_config(duthost, config, config_backup):
+    logger.info("Backup {} to {} on {}".format(
+        config, config_backup, duthost.hostname))
+    duthost.shell("cp {} {}".format(config, config_backup))
+
+
+def restore_config(duthost, config, config_backup):
+    logger.info("Restore {} with {} on {}".format(
+        config, config_backup, duthost.hostname))
+    duthost.shell("mv {} {}".format(config_backup, config))
+
+
+def get_running_config(duthost):
+    return json.loads(duthost.shell("sonic-cfggen -d --print-data")['stdout'])
+
+
+def reload_minigraph_with_golden_config(duthost, json_data):
+    duthost.copy(content=json.dumps(json_data, indent=4), dest=GOLDEN_CONFIG)
+    config_reload(duthost, config_source="minigraph", safe_reload=True)
+
+
+@pytest.fixture(scope="module")
+def setup_env(duthosts, rand_one_dut_hostname, golden_config_exists_on_dut):
+    """
+    Setup/teardown
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # Backup configDB
+    backup_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Backup Golden Config if exists.
+    if golden_config_exists_on_dut:
+        backup_config(duthost, GOLDEN_CONFIG, GOLDEN_CONFIG_BACKUP)
+
+    # Reload test env with minigraph
+    config_reload(duthost, config_source="minigraph", safe_reload=True)
+    running_config = get_running_config(duthost)
+
+    yield running_config
+
+    # Restore configDB after test.
+    restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
+    # Restore Golden Config after test, else cleanup test file.
+    if golden_config_exists_on_dut:
+        restore_config(duthost, GOLDEN_CONFIG, GOLDEN_CONFIG_BACKUP)
+    else:
+        duthost.file(path=GOLDEN_CONFIG, state='absent')
+
+    # Restore config before test
```

Comment:
>before [](http://example.com/codeflow?start=21&length=6)

You mean "after this test"?

Reply:
Mean restore to the config when the test is not started.